pkg/sypgp/sypgp.go

```diff
@@ -614,6 +614,17 @@ func SearchPubkey(search, keyserverURI, authToken string) error {
 
 // FetchPubkey pulls a public key from the Key Service.
 func FetchPubkey(fingerprint, keyserverURI, authToken string, noPrompt bool) (openpgp.EntityList, error) {
+	if len(fingerprint) < 8 {
+		return nil, fmt.Errorf("not a valid key lenth: %s: must be over 8 chars", fingerprint)
+	}
+
+	// Only chop the fingerprint if its less then 40 chars
+	if len(fingerprint) < 40 {
+		// Harvest the last 8 chars of the fingerprint string
+		sylog.Debugf("Getting the last 8 chars from: %s", fingerprint)
+		fingerprint = fingerprint[len(fingerprint)-8:]
```

Comment:
not sure I agree with this.

The user is providing a specific input, and in an effort to be "user friendly" we would be _discarding_ part of the input.

The problem with this is that we would be making the search less specific, potentially returning results the user was NOT looking for.

For clarity, the fingerprint a 160-bit hash of part of the public key; the long key ID is the lowest-order 64 bits of this hash; the short key ID is the lowest-order 32 bits of this hash. In hexadecimal this means the fingerprint is 40 digits, the short key ID is 8 digits and the long key ID is 16 digits.

Consider a "highly hypothetical" case of two keys sharing the same _short key ID_. If the user intended to type a fingerprint, but misses a digit, so instead of entering 40 digits enters 39, this code would remove 31 of those digits and keep only 8. Instead of getting an error, the user would get zero, one or two results, _depending on the keyserver it's talking to_. In a normal situation the user would still get the intended key, but this is highly dependent on external factors. What if the user gets the _other_ key though? Meaning, with the original query the user would get a "not found message", but because of the typo they get something.

I would argue this is _less_ desirable than the error.

My take is that this code should be strict about checking for 40, 16 or 8 digits. I would go a step further and refuse to use short key IDs, as collisions in short key IDs are in fact not "highly hypothetical" (http://www.asheesh.org/note/debian/short-key-ids-are-bad-news.html).

Reply:
So it seems like the issue is really with the SCS API right?  Maybe this should be handled the same way git handles hashes on the KeyStore side.  It will accept a fingerprint of any length and then will either return the key or barf because there is a collision and the user must provide additional chars.